imports/api/users.js

```diff
@@ -61,24 +61,49 @@ _.extend(User.prototype, {
     else return false
   },
   getImageUrl: function () {
-    if (this.profile.profileThumbnail) {
+    if (this.profile.profileImage) {
       const settings = Settings.findOne()
-      if ( settings && settings.storageType === 'AWS' && !this.profile.profileImage.endsWith('/image')) {
+      if ( settings && settings.storageType === 'AWS' && this.profile.profileImage.startsWith('https://' + settings.AWS_bucket + '.s3-' + settings.AWS_region + '.amazonaws.com') && !this.profile.profileImage.endsWith('/image')) {
         Meteor.call('users.updateProfileImage', this.profile.profileImage + '/image', settings.storageType)
-      
       }
+      if ( settings && settings.storageType === 'AWS' && this.profile.profileThumbnail.startsWith('https://' + settings.AWS_bucket + '.s3-' + settings.AWS_region + '.amazonaws.com') && !this.profile.profileThumbnail.endsWith('/thumbnail')) {
```

Comment:
I really don't think this needs to check the current storage type. There are 2 scenarios:

1) The profile pics were uploaded in the old DB (which starts with the string that I gave in the email), in which case the profile and thumb properties are not correctly set to URLs and we should update the profile to set them to URLs. After this, only case 2) should happen.

2) The profile pic and thumb properties were created with updated code, so they are already set to URLs, and we can just return those URLs.

Reply:
Ah I see. The reason I was checking the settings is to check the AWS_bucket and AWS_region properties when scanning the start of the image url. Perhaps this can be changed to a regular expression

Reply:
Nah, I think since this really is specific to the one database, you can just put the actual string. Once the db is updated, we'll just remove that change.